### ISSUE TITLE
Allow reading config from tox.ini / setup.cfg?

### ISSUE THREAD

**Issue by aldanor** [ionelmc/pytest-benchmark#26]
Seeing as there's a growing list of options that barely fits one help screen, maybe it would make sense to think about reading the default from config file same as most other tools do (tox itself, coverage, pytest, all sorts of linters)?

So instead of passing a bajillion of command line arguments, you could just add a section in your `tox.ini` like so (just making these up):

```ini
[benchmark]
columns = min, max
precision = auto
report = term, json
json = foo.json
```

**Comment by ionelmc** [comment]
Just for refference, the complete list of current options:
```ini
[benchmark]
autosave = false
calibration_precision = 10
compare =
compare_fail =
disable_gc = false
group_by = group
histogram =
json =
max_time = 1.0
min_rounds = 5
min_time = 0.000005
only = false
save =
save_data =
skip = false
sort = min
storage =
timer = time.perf_counter
verbose = false
warmup = false
warmup_iterations = 100000
# new options
columns = min, max
precision = auto
report = term, json
```
```

**Comment by aldanor** [comment]
Whoah, this definitely deserves a config file section!

**Comment by ionelmc** [comment]
Maybe we should use some namespaces, eg:
```ini 
[benchmark:report]
sort = min
compare = 
compare_fail = 

[benchmark:report:term]
class = mod.MyCustomRenderer
precision = auto
columns  = min, max
```

**Comment by anarcat** [comment]
+1

**Comment by ionelmc** [comment]
Another idea is to use environment variables

**Comment by aldanor** [comment]
@ionelmc Haven't been using pytest-benchmark, now the time has come again, I've searched for ways to configure it and... found my own ticket :)

Wondering if there are any news on this front, or help needed?

**Comment by ionelmc** [comment]
@aldanor ah sorry I haven't done anything on this yet. Do you want to help with a PR?

**Comment by aldanor** [comment]
Yet another thought... `pyproject.toml`? (seeing as more and more libraries and projects are moving towards it these days)

**Comment by ionelmc** [comment]
Worth checking out if tox/pytest will move to toml (there was some discussion about that but I forgot where) and just do what they do.

**Comment by matham** [comment]
I'm a little confused about whether there's currently some support for reading config from a file. E.g. https://github.com/ionelmc/pytest-benchmark/issues/27 and in the docs:

```
--benchmark-enable
   Forcibly enable benchmarks. Use this option to override –benchmark-disable (in case you have it in pytest configuration).
```

seem to indicate there's already a way to read from pytest configuration files since it's causally mentioned!?

**Comment by ionelmc** [comment]
Look no further: https://docs.pytest.org/en/stable/customize.html (eg: `addopts = --benchmark-something`). Tho I don't think it's a good idea to have `--benchmark-enable` in the config file - you'd never be able to disable the benchmarks.

**Comment by ionelmc** [comment]
There is none currently, just addopts. Is there's something wrong with addopts? You can have those over multiple lines afaik. It's a bit noisy, having to prefix everyting with `--benchmark-` but it feels to me like there's something wrong with the defaults for some options. Tell me more.

**Comment by ofek** [comment]
Yes, there is something very wrong with that but the issue is at a higher level. Benchmarks reside in their own virtual environment with a separate execution model from tests so that option you speak of would apply globally and when running `pytest` in, say the test environment, will error with unknown options.
